### PR TITLE
Container image optimisations

### PR DESCRIPTION
**Pull request by cmgrote**
Egeria image changes:

- Fixes permissions on the Egeria image to run as non-root "egeria" user, and moves all contents into `/opt/egeria/...` rather than the root filesystem

Apache Ranger image changes:

- Moves Apache Ranger password setup into configmap, rather than hard-coding
- Uses postgresql database backend rather than mariadb
- Uses initContainer approach for ensuring database is running before starting (removing need for dockervize)
- Applies multi-stage build to obtain a minimal runtime image of ~580MB (vs previous 4.4GB)
- Executes Ranger and Solr processes as dedicated non-root users (unfortunately must run the container as root still due to the tightly-coupled setup scripts of Ranger: for creating various log directories and deploying database entries)

Also adds an Apache Solr pod based on the Docker Official image (as basis for future optimisation under #1041).

### CODE REVIEW

**Comment by askb** [comment]
recheck

File: open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile [hunk 1]
```diff
@@ -1,62 +1,26 @@
-# SPDX-License-Identifier: CC-BY-4.0 
-# Copyright Contributors to the Egeria project. 
-# ---
-# Docker configuration file for Ranger
-
-# Let's use UBUNTU
-FROM            ubuntu:18.04
-
-# Image metadata following http://label-schema.org/rc1/
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile]: I notice this change has dropped the metadata labels.
One could discuss how relevant they are, though we should probably have them in all or images, or none
Some tips at https://medium.com/@chamilad/lets-make-your-docker-image-better-than-90-of-existing-ones-8b1e5de950d
We could perhaps replace the version with a simple build date, as that is easy to automate.
Probably the best approach is to simply open up a new issue to correct the tags, and allow this change to go through.

File: open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml [hunk 2]
```diff
@@ -135,11 +146,9 @@ image:
     name: infosvr
     tag: v11.7.0.2
 
-  configure:
-    name: egeria-configure
-
-  nameOverride: ""
-  fullnameOverride: ""
+# These should not be embedded under 'image' (?)
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml]: correct. (they are only used to override the base chart name)

File: open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile [hunk 3]
```diff
@@ -66,109 +30,67 @@ RUN mvn -Pall -DskipTests=true clean compile package install assembly:assembly
...
-RUN sed -i "s|SOLR_DOWNLOAD_URL=|SOLR_DOWNLOAD_URL=http://www.mirrorservice.org/sites/ftp.apache.org/lucene/solr/${solrver}/solr-${solrver}.tgz|" install.properties
-
-#Update path to include java, so that when we run the container it is available
-#default shell isn't bash :-(
-
-#RUN echo "export JAVA_HOME=/usr/lib/jvm/default-java" >> /root/.bashrc
-RUN echo "export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64" >> /root/.profile
-RUN echo "export PATH=${JAVA_HOME}/bin:$PATH" >> /root/.profile
-RUN chmod +x /root/.profile
+RUN sed -i "s|SOLR_DOWNLOAD_URL=|SOLR_DOWNLOAD_URL=http://www.mirrorservice.org/sites/ftp.apache.org/lucene/solr/7.7.1/solr-7.7.1.tgz|" install.properties
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile]: the solr version can be a little fragile - usually the last point release is left. Just a note.

**Comment by planetf1** [comment]
A useful set of changes -- thanks. I think we should open an issue to consolidate/discuss image metadata. I'll pull and build to test.

**Comment by cmgrote** [comment]
Agreed — consistent image metadata is needed.  I’m happy to be assigned that in a new PR.

File: open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile (hunk 1 above)
Comment (planetf1) [reply open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile]: Opened https://github.com/odpi/egeria/issues/1048 to follow up

**Comment by planetf1** [comment]
@cmgrote failed rat check

File: open-metadata-resources/open-metadata-deployment/docker/ranger/dist/ranger-docker-start.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/docker/ranger/dist/ranger-docker-start.sh]: Missing license declaration

File: open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile [hunk 5]
```diff
@@ -66,109 +30,77 @@ RUN mvn -Pall -DskipTests=true clean compile package install assembly:assembly
...
+RUN sed -i "s|^SQL_CONNECTOR_JAR=/usr/share/java/mysql-connector-java.jar|SQL_CONNECTOR_JAR=/usr/share/java/postgresql-jdbc.jar|g" install.properties
 RUN sed -i "s|^audit_solr_urls=|audit_solr_urls=http://localhost:6083/solr/ranger_audits|g" install.properties
-RUN sed -i "s|^rangerAdmin_password=|rangerAdmin_password=${rootpw}|g" "${dadir}/install.properties"
-RUN sed -i "s|^rangerTagsync_password=|rangerTagsync_password=${rootpw}|g" "${dadir}/install.properties"
-RUN sed -i "s|^rangerUsersync_password=|rangerUsersync_password=${rootpw}|g" "${dadir}/install.properties"
-RUN sed -i "s|^keyadmin_password=|keyadmin_password=${rootpw}|g" "${dadir}/install.properties"
 
-# Expose web ports
-EXPOSE          6080 6182 6083 6183
+# Create minimal runtime image
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile]: One example here ... we could probably leave this issue open -- I think we could probably look through how a consumer may use the metadata. One specific example I was thinking of was the build date (even though we have the git history on disk)

Comment (cmgrote) [reply open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile]: Per #1048 the build date is already tracked by Docker itself (as are the exposed ports)...  I'm not keen on tracking the same metadata in duplicate ways that then either must be kept in sync or risk confusing consumers as to which of the conflicting metadata is correct (?)

Comment (planetf1) [reply open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile]: will close as review and consider discussion in #1048

**Comment by planetf1** [comment]
testing shortly and will merge later if good.

**Comment by planetf1** [comment]
changes look good. Was able to deploy, access ranger UI, manipulate policies. Updating and merging

**Pull request merged**